libsolidity/codegen/LValue.cpp

```diff
@@ -144,6 +144,42 @@ void MemoryItem::setToZero(SourceLocation const&, bool _removeReference) const
 	m_context << Instruction::POP;
 }
 
+
+ImmutableItem::ImmutableItem(CompilerContext& _compilerContext, VariableDeclaration const& _variable):
+	LValue(_compilerContext, _variable.annotation().type), m_variable(_variable)
+{
+	solAssert(_variable.immutable(), "");
+}
+
+void ImmutableItem::retrieveValue(SourceLocation const&, bool) const
+{
+	solUnimplementedAssert(m_dataType->isValueType(), "");
+	solAssert(!m_context.runtimeContext(), "Tried to read immutable at construction time.");
```

Comment:
It is not implemented in the code generator: It would at least require properly initializing the immutable at the very start, but I would say that people have the impression that immutables can only be set exactly once and cannot be used before. To simplify the analysis, we just disallow reading at all in construction context.

Reply:
Ok, what's with the tests then? Are they going to be removed?

Reply:
I wanted to wait until we have the type checks in place.

Reply:
Which PR/issue has the type checks?

Reply:
https://github.com/ethereum/solidity/pull/8475